packages/react/src/ActionList/List.tsx

```diff
@@ -33,15 +33,20 @@ export const List = React.forwardRef<HTMLUListElement, ActionListProps>(
 
     /** if list is inside a Menu, it will get a role from the Menu */
     const {
-      listRole,
+      listRole: listRoleFromContainer,
       listLabelledBy,
       selectionVariant: containerSelectionVariant, // TODO: Remove after DropdownMenu2 deprecation
-      enableFocusZone,
+      enableFocusZone: enableFocusZoneFromContainer,
     } = React.useContext(ActionListContainerContext)
 
     const ariaLabelledBy = slots.heading ? slots.heading.props.id ?? headingId : listLabelledBy
-
+    const listRole = role || listRoleFromContainer
     const listRef = useProvidedRefOrCreate(forwardedRef as React.RefObject<HTMLUListElement>)
+
+    let enableFocusZone = false
+    if (enableFocusZoneFromContainer !== undefined) enableFocusZone = enableFocusZoneFromContainer
+    else if (listRole) enableFocusZone = ['menu', 'menubar', 'listbox'].includes(listRole)
```

Comment:
Non-blocking: I see we have have `menubar`, I am wondering if there'd be a case where `role=menubar` is used with ActionList 🤔

Reply:
We don't have one yet, but I think it's possible. The [issue linked to menu + menubar pattern](https://arc.net/l/quote/brrnmqyb), so I included it just in case